### PR TITLE
Fix up elevators and their controls

### PR DESCRIPTION
**Pull request by AlexMooney**
### Summary

```SUMMARY: Features "Creatures can ride along elevators with you"```

#### Purpose of change

Elevator controls are fragile and exploitable.  They can easily be smashed up and when you arrive at the smashed controls, you're unable to get back out.  You can also teleport away from enemies as long as you can hit the elevator controls.

#### Describe the solution

- If there are creatures in an elevator with you, they come along for the ride.
- If there are creatures in an elevator at the destination, push them out of the elevator before arrival.
- Elevator controls are really hard to break.  This should go away if & when climbing up the elevator shaft is possible.

#### Other considerations

- Monsters can be pushed through walls as easily as they're pushed out of the elevator doors.
- Sometimes you get messages about slipping past monsters when using the elevator.  I don't think these changes caused this.
- ~Sometimes you arrive on the other side of a wall from the controls.  I don't think these changes caused this either.~ Fixed!
- Items are left where they are on the map, so the illusion isn't complete.

#### Testing

Went up and down a lab elevator with some zombie scientists with me or at the destination.  Tried it with experimental Z levels on and off and it seems to work in both cases.

I saw one debug error about a monster seeking the stairs as I was testing this, but the save was before a recompile, so I'm not sure if the bug is still there.  Probably needs some more testing.

### CODE REVIEW

File: src/iexamine.cpp [hunk 1]
```diff
@@ -744,6 +744,32 @@ void iexamine::elevator( player &p, const tripoint &examp )
...
+            if( ms_to_omt_copy( g->m.getabs( critter.pos() ) ) == ms_to_omt_copy( g->m.getabs( examp ) ) ) {
+                // they're on the elevator and come along
+                for( const tripoint &candidate : closest_tripoints_first( 10, p.pos() ) ) {
+                    if( g->m.ter( candidate ) == ter_id( "t_elevator" ) &&
+                        !g->critter_at( candidate ) ) {
+                        critter.setpos( candidate );
+                        break;
+                    }
+                }
+            } else if( ms_to_omt_copy( g->m.getabs( critter.pos() ) ) == ms_to_omt_copy( g->m.getabs(
```
Comment (AlexMooney) [inline src/iexamine.cpp]: This line is too long.  Maybe I'll do an assignment after L749 to shorten these if statements.

**Comment by AlexMooney** [comment]
To really do this right, I'd need to iterate over all creatures twice:

1. Push the destination floor creatures out of the way.
2. `g->vertical_move( movez, false );`
3. Bring the departing floor creatures along.

By doing these out of order to save a for loop, it's possible to block the elevator by filling up the destination floor.  I'll take another pass at fixing this later this week.

**Comment by Amoebka** [comment]
There should be no destination floor creatures. If the elevator is on one floor then the other floor is empty shaft zombies can't stand on.

**Comment by AlexMooney** [comment]
> There should be no destination floor creatures. If the elevator is on one floor then the other floor is empty shaft zombies can't stand on.

This is conceptually true, but right now the elevator exists simultaneously on both floors and using the controls teleports the player two floors up or two floors down.  I'm trying to make the approximation of an elevator better, but I'm not trying to make a moving elevator car inside of an elevator shaft.  Before that could be considered, the maps would need to be reworked so that the elevators are vertically aligned.  Lateral teleportation is involved too.

**Comment by AlexMooney** [comment]
I've tested this by filling both the starting elevator and the destination elevator, and it works as I intend.  I think this is good to go.

File: src/iexamine.cpp [hunk 2]
```diff
@@ -743,7 +743,51 @@ void iexamine::elevator( player &p, const tripoint &examp )
         return;
     }
     int movez = ( examp.z < 0 ? 2 : -2 );
+
+    tripoint original_floor_omt = ms_to_omt_copy( g->m.getabs( examp ) );
+    tripoint new_floor_omt = original_floor_omt + tripoint( 0, 0, movez );
```
Comment (ZhilkinSerg) [inline src/iexamine.cpp]:
```suggestion
    tripoint new_floor_omt = original_floor_omt + tripoint( point_zero, movez );
```

**Pull request merged**